### DISCUSSION TITLE
关于输入法界面的建议

### DISCUSSION THREAD

**Discussion by jack6th** [rime/home#992] [Ideas]
在Windows中，拼音输入法的界面大致有以下三种。
第一种，以Windows 10的微软拼音输入法为代表。特点是光标处显示拼音，优点是候选项和光标靠得比较紧，缺点是光标处的拼音会对用户形成干扰。
![ms_8bit](https://user-images.githubusercontent.com/34613100/149069515-1591b51b-c809-4387-8840-42c1d0175ac5.png)
第二种，以小狼毫为代表。特点是光标处不显示，在输入法界面第一行显示拼音，第二行开始显示候选项，缺点是第一候选项和光标离得比较远。
![rime_8bit](https://user-images.githubusercontent.com/34613100/149071836-58bd7bce-99bd-48de-8a87-3e08ba0e877a.png)
第三种，以谷歌拼音输入法为代表（截图为v2.7版）。特点是光标处显示第一候选项，该候选项的字体和大小与原程序中一样，输入体验接近于直接输入英文。缺点是在某些程序中输入时，光标处嵌入的字体和大小与原程序不一致。
![google_8bit](https://user-images.githubusercontent.com/34613100/149072878-ea36cd3b-b356-4640-b683-fa7a6a514c40.png)
微软拼音输入法不能自定义字体和背景色，在暗黑主题之下，输入法背景也是白色的，不太协调。
谷歌拼音输入法可以为拼音和候选项设置不同的字体和大小，不过各种皮肤的右上角都有一个Google的图标，破坏了界面的简洁性，更遗憾的是该软件已经停止更新。
希望小狼毫能借鉴谷歌拼音输入法的嵌入式设计，提供一个光标处直接显示第一候选的界面选项。

**Comment by LEOYoon-Tsaw** [comment]
有了

**Comment by MokOopsing** [comment]
不是没有，是你不知道（不会用）

**Reply by MokOopsing** [reply]
```yaml
#weasel.custom.yaml
patch:
  style:
    inline_preedit: true
    preedit_type: preview #composition
```

**Comment by jack6th** [comment]
复制了楼上的代码，的确可以，效果很好。
有没有包含所有的（或最常用的）功能选项的weasel.custom.yaml文件的样板，初始状态是全部注释掉的，用户可以根据需要打开。小狼毫主要使用文本而不是GUI来配置，这样做能降低配置难度。

从空白开始编辑，需要在网上寻找代码，而网上大多都是一些代码片段，不完整。楼上的这段代码中preedit_type选项在Wiki页面的《说明书》、《定制指南》、《雪齋的文檔》中都没有。

**Reply by LEOYoon-Tsaw** [reply]
小狼毫好像没有，鼠鬚管倒是我做了一個帶UI的配置器生成squirrel.yaml

**Reply by WilburTsai** [reply]
win10 ,  改preview毫无变化。

搜索所有相关文件夹，文件中出现preedit_type的，仅有weasel.yaml。

也无处查找，preedit_type后面可以填写的合法参数一共有哪几个。

**Reply by pdog18** [reply]
> win10 , 改preview毫无变化。
> 
> 搜索所有相关文件夹，文件中出现preedit_type的，仅有weasel.yaml。
> 
> 也无处查找，preedit_type后面可以填写的合法参数一共有哪几个。

小白用这个吧 [助手](https://pdog18.github.io/rime-soak/#/theme)
